saleor/graphql/payment/mutations.py

```diff
@@ -105,6 +106,21 @@ def clean_payment_amount(cls, info, checkout_total, amount):
                 }
             )
 
+    @classmethod
+    def validate_gateway(cls, gateway_id, currency):
+        available_gateways = get_plugins_manager().list_payment_gateways(
+            currency=currency
+        )
+        if not any([gateway["id"] == gateway_id for gateway in available_gateways]):
+            raise ValidationError(
+                {
+                    "gateway": ValidationError(
+                        "This gateway does not support checkout currency.",
```

Comment:
Wouldn't it be good to return the name of the gateway here?

Reply:
I think it doesn't matter, the getaway is provided as input to the mutation. But to be more specific I can add gateway name there.